src/ansible_ii_leader.c

```diff
@@ -328,9 +328,129 @@ static void ii_slew_txo(i2c_follower_t* follower, uint8_t track, uint16_t slew)
 	}
 }
 
+static void ii_mode_disting_ex(i2c_follower_t* follower, uint8_t track, uint8_t mode) {
+	if (mode > follower->mode_ct) return;
+	follower->active_mode = mode;
+}
+
+static s16 calculate_note(s16 dac_value, s8 octave) {
+	s32 note = (dac_value * 240) / 16384;
+	note = note / 2 + (note & 1) + 36 + octave * 12;
+	return note;
+}
+
+static void ii_tr_disting_ex(i2c_follower_t* follower, uint8_t track, uint8_t state) {
+	uint8_t d[4] = { 0 };
+	s16 note = calculate_note(dac_get_value(track), follower->oct);

```

Comment:
This reminds me that sending the current `dac_get_value` is potentially out of tune if the user has a custom tuning table saved to correct for the module's DAC offsets. The JF trigger function does the same thing. I think it's not a huge deal right now but we should maybe have another issue to buffer the requested semitone value for each track -- this would also let you skip the reverse pitch -> note calculation.

Reply:
yeah, i think it would be useful if we stored the note id in addition to pitch. additionally, i think we should also store the DAC value without the calibration offsets applied, if we decide to support microtonal scales at some point.